### PR TITLE
SWATCH-288: Update Jenkins pipeline to use containers as nodes

### PR DESCRIPTION
**Pull request by lindseyburnett**
Instead of using "fedora" or "fedora35" node on the SMQE jenkins instance for our **continuous-integration/jenkins/pr-merge** check for each PR, have jenkins spin up a new container to use as a build node.

"Testing"/Verification

If you click "Details" in the checks section of this PR, and then navigate to the Jenkins instance[0]  (top right corner....icon that kind of looks like logout), and then open the console output and View Full Log....you'll be able to see near the top that an agent was provisioned based on a pod spec yaml. 

If the job has been ran recently, you can actually click into the linked agent name that got spun up and see some details within Jenkins.  You can also take that name and search for it in the openshift console [1], where you can view the node and access the terminal.

[0] https://stage-jenkins-csb-smqe.apps.ocp4.prod.psi.redhat.com/job/Subscriptions%20PR%20Check/
[1] https://console-openshift-console.apps.ocp4.prod.psi.redhat.com/topology/ns/jenkins-csb-smqe?view=graph

### CODE REVIEW

**Comment by lindseyburnett** [comment]
close to stop the auto build spam

File: Jenkinsfile [hunk 1]
```diff
@@ -1,75 +1,62 @@
...
-                        catch (org.jenkinsci.plugins.workflow.steps.FlowInterruptedException e) {
-                            // "rethrow" as something retry will actually retry, see https://issues.jenkins-ci.org/browse/JENKINS-51454
-                            if (e.causes.find { it instanceof org.jenkinsci.plugins.workflow.steps.TimeoutStepExecution$ExceededTimeout } != null) {
-                                error("Timeout waiting for SonarQube results")
-                            }
-                        }
-                    }
-                }
+        kubernetes {
+            label 'swatch' // this value + unique identifier becomes the label
```
Comment (kahowell) [inline Jenkinsfile:5]: becomes the pod name?

File: Jenkinsfile [hunk 2]
```diff
@@ -1,75 +1,62 @@
...
+           }
+       }
 
+       stage('Upload PR to SonarQube') {
+           when {
+               changeRequest()
+           }
+           steps {
+               withSonarQubeEnv('sonarcloud.io') {
+                   sh "./gradlew  --no-daemon  sonarqube -Duser.home=/tmp -Dsonar.host.url=${SONAR_HOST_URL} -Dsonar.login=${SONAR_AUTH_TOKEN} -Dsonar.pullrequest.key=${CHANGE_ID} -Dsonar.pullrequest.base=${CHANGE_TARGET} -Dsonar.pullrequest.branch=${BRANCH_NAME} -Dsonar.organization=rhsm -Dsonar.projectKey=rhsm-subscriptions"
```
Comment (kahowell) [inline Jenkinsfile:34]: Nitpick: extraneous space character throughout between `gradlew` and `--no-daemon`

File: Jenkinsfile [hunk 3]
```diff
@@ -1,75 +1,62 @@
...
+                resourceLimitCpu '6'
+                resourceRequestMemory '2Gi'
+                resourceLimitMemory '6Gi'
             }
+
+            defaultContainer 'openjdk11'
         }
     }
+    stages {
+       stage('Build') {
```
Comment (kahowell) [inline Jenkinsfile:22]: Probably worth commenting that what this effectively invokes with a comment, consider changing stage name too; suggestion: `Build/Test/Lint`

**Review by kahowell — CHANGES_REQUESTED** [review]

**Review by kahowell — APPROVED** [review]

**Pull request merged**